### PR TITLE
M2P-280 Code refactoring for 3rd-party module Aheadworks StoreCredit

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
Code refactoring for 3rd-party module Aheadworks StoreCredit

Fixes: https://boltpay.atlassian.net/browse/M2P-280

#changelog M2P-280 Code refactoring for 3rd-party module Aheadworks StoreCredit

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Comment by danac-gs** [comment]
Can you add some unit tests?

**Comment by johnmccombs1** [comment]
@danac-gs I was able to add a unit test for the class ThirdPartyModules/Aheadworks/StoreCredit.php as in this PR https://github.com/BoltApp/bolt-magento2/pull/928. However, Vitaly mentioned that it's not useful 

@danac-gs @vitaliyreznikov @matt-thomason Thought on this?

File: ThirdPartyModules/Aheadworks/StoreCredit.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        }
+    }
+
+    /**
+     * Get Aheadworks store credit for the user
+     *
+     * @param int $customerId Logged in customer ID
+     * @return float User store credit
+     */
+    private function getAheadworksStoreCredit($customerId)
```
Comment (dumega) [inline ThirdPartyModules/Aheadworks/StoreCredit.php:108]: @johnmccombs1 Do we need such a one-line function?

**Review by danac-gs — APPROVED** [review]

File: ThirdPartyModules/Aheadworks/StoreCredit.php (hunk 1 above)
Comment (johnmccombs1) [reply ThirdPartyModules/Aheadworks/StoreCredit.php]: @dumega I've moved it into the collectDiscounts method. Could you please recheck?

**Review by danac-gs — APPROVED** [review]

**Review by dumega — APPROVED** [review]
LGTM, please fix the conflict.

**Comment by johnmccombs1** [comment]
@dumega I've fixed the conflict

**Review by dumega — APPROVED** [review]

**Review by danac-gs — APPROVED** [review]

**Pull request merged**